services/ExportService.php

```diff
@@ -89,10 +89,18 @@ public function download(array $settings)
             fputcsv($export, $this->parseColumns($settings));
 
             // Loop trough data
-            foreach ($data as $element) {
+            foreach ($data as $row) {
+
+                // fetch element in case of element id
+                if (is_numeric($row)) {
+                    $criteria = $this->_service->setCriteria($settings);
+                    $criteria->id = $row;
+                    $element = $criteria->first();
+                    $row = $element;
+                }
```

Comment:
The contents of this whole block can be replaced by `$row = craft()->elements->getElementById($row, $settings['type']);`